godtossing/Pos/Ssc/GodTossing/Workers.hs

```diff
@@ -419,3 +431,46 @@ waitUntilSend msgTag epoch slMultiplier = do
                 ttwMillisecond
                 msgTag
         delay timeToWait
+
+----------------------------------------------------------------------------
+-- Security check
+----------------------------------------------------------------------------
+
+checkForIgnoredCommitmentsWorker
+    :: forall ctx m.
+       SscMode SscGodTossing ctx m
+    => (WorkerSpec m, OutSpecs)
+checkForIgnoredCommitmentsWorker = localWorker $ do
+    counter <- newTVarIO 0
+    void $ onNewSlot True (checkForIgnoredCommitmentsWorkerImpl counter)
+
+-- This worker checks whether our commitments appear in blocks. This
+-- check is done only if we actually should participate in
+-- GodTossing. It's triggered if there are
+-- 'mdNoCommitmentsEpochThreshold' consequent epochs during which we
+-- had to participate in GodTossing, but our commitment didn't appear
+-- in blocks.
+checkForIgnoredCommitmentsWorkerImpl
+    :: forall ctx m. (SscMode SscGodTossing ctx m)
+    => TVar Word -> SlotId -> m ()
+checkForIgnoredCommitmentsWorkerImpl counter SlotId {..}
+    -- It's enough to do this check once per epoch near the end of the epoch.
+    | getSlotIndex siSlot /= 9 * fromIntegral blkSecurityParam = pass
+    | otherwise =
+        recoveryCommGuard $
+        whenM (shouldParticipate siEpoch) $ do
+            ourId <- getOurStakeholderId
+            globalCommitments <-
+                getCommitmentsMap . view gsCommitments <$> gtGetGlobalState
+            unless (ourId `HM.member` globalCommitments) $ do
+                -- `modifyTVar'` returns (), hence not used
+                newCounterValue <-
+                    atomically $ do
+                        !x <- succ <$> readTVar counter
+                        x <$ writeTVar counter x
+                when (newCounterValue > mdNoCommitmentsEpochThreshold) $ do
```

Comment:
Shouldn't you reset this counter at some point?

Reply:
I should.